src/attachment.rs

```diff
@@ -97,16 +107,23 @@ impl Attachment {
             .unwrap_or("file")
             .to_owned();
 
-        let file = File::open(path).await?;
+        let file = File::open(path.clone()).await?;
```

Comment:
The usual way I handle this is, at the top of the function, writing something like:

```rust
let path = path.as_ref();
```

`path` becomes a `&Path`, which also impls `AsRef<Path>`, so it can be passed to both functions without a clone.

Reply:
sick thank you! constantly learning new things